### PR TITLE
feat(fcm): Add image in notification support

### LINKED ISSUE #598: FR: add image option in notification
There are new options for image in notification : https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages?authuser=0#notification

### PR DESCRIPTION
**Pull request by lahirumaramba**
### Discussion

  * Add image in notification support

### Testing

  * Add unit tests and modified integration test to reflect this change

### API Changes

  * In Firebase Cloud Messaging, added 'image' field in the general Notification class, the AndroidNotification class, and the ApnsFcmOptions class.

RELEASE NOTE: Added new APIs for specifying an image URL in notifications.

Resolves #598

### CODE REVIEW

File: test/unit/messaging/messaging.spec.ts [hunk 1]
```diff
@@ -2334,6 +2334,37 @@ describe('Messaging', () => {
       });
     });
 
+    const invalidImages = ['a', 'foo', 'image.jpg'];
```
Comment (hiranya911) [inline test/unit/messaging/messaging.spec.ts]: Also test for empty string

File: test/unit/messaging/messaging.spec.ts [hunk 2]
```diff
@@ -2334,6 +2334,37 @@ describe('Messaging', () => {
...
+            },
+          },
+        };
+        expect(() => {
+          messaging.send(message);
+        }).to.throw('android.notification.image must be a valid URL string');
+      });
+    });
+
+    invalidImages.forEach((image) => {
```
Comment (hiranya911) [inline test/unit/messaging/messaging.spec.ts]: Lets move this somewhere else in this file. This section seems to be all about android config tests.

File: src/messaging/messaging-types.ts [hunk 3]
```diff
@@ -59,6 +59,7 @@ export interface MulticastMessage extends BaseMessage {
 export interface Notification {
   title?: string;
   body?: string;
+  image?: string;
```
Comment (hiranya911) [inline src/messaging/messaging-types.ts]: We called these `imageUrl` in Java. Might want to do the same here.

Comment (lahirumaramba) [reply src/messaging/messaging-types.ts]: @hiranya911 
Do you think it is sufficient to change just the public API `index.d.ts` here or should we use the same parameter name `imageUrl` everywhere in our code?
